src/PoolWatcher.ts

```diff
@@ -350,115 +344,141 @@ export class PoolWatcher extends TypedEmitter<PoolWatcherEvents> {
 
     const upkeepSuccessfulFilter = this.watchedPool.keeperInstance.filters.UpkeepSuccessful(this.poolAddress);
 
-    const scheduleStateCalculation = async () => {
-      const [
-        lastPriceTimestampEthersBN,
-        appropriateIntervalIdBefore
-      ] = await Promise.all([
-        attemptPromiseRecursively({ promise: () => this.poolInstance.lastPriceTimestamp() }),
-        attemptPromiseRecursively({ promise: () => this.watchedPool.committerInstance.getAppropriateUpdateIntervalId() })
-      ]);
+    if (!this.ignoreEvents[EVENT_NAMES.COMMITMENT_WINDOW_ENDING]) {
+      const scheduleStateCalculation = async () => {
+        const [
+          lastPriceTimestampEthersBN,
+          appropriateIntervalIdBefore
+        ] = await Promise.all([
+          attemptPromiseRecursively({ promise: () => this.poolInstance.lastPriceTimestamp() }),
+          attemptPromiseRecursively({ promise: () => this.watchedPool.committerInstance.getAppropriateUpdateIntervalId() })
+        ]);
 
-      const { frontRunningInterval, updateInterval } = this.watchedPool as WatchedPool;
+        const { frontRunningInterval, updateInterval } = this.watchedPool as WatchedPool;
 
-      const lastPriceTimestamp = lastPriceTimestampEthersBN.toNumber();
-      const commitmentWindowEnd = frontRunningInterval < updateInterval
+        const lastPriceTimestamp = lastPriceTimestampEthersBN.toNumber();
+        const commitmentWindowEnd = frontRunningInterval < updateInterval
         // simple case
-        ? lastPriceTimestamp + updateInterval - frontRunningInterval
+          ? lastPriceTimestamp + updateInterval - frontRunningInterval
         // complex case, multiple update intervals within frontRunningInterval
-        : lastPriceTimestamp + updateInterval;
+          : lastPriceTimestamp + updateInterval;
 
-      // calculate the time at which we should wait until to calculate expected pool state
-      const waitUntil = commitmentWindowEnd - this.commitmentWindowBuffer;
+        // calculate the time at which we should wait until to calculate expected pool state
+        const waitUntil = commitmentWindowEnd - this.commitmentWindowBuffer;
```

Comment:
whats the buffer here?

Reply:
this is the config for when to calculate state and emit this event (ie 10 seconds before end of commitment window)